Adapt IDL harness test for battery status to Promise-returning factory

dontcallmedom opened the pull request.

dontcallmedom committed:
Adapt IDL harness test for battery status to Promise-returning factory

tobie commented:
Reviewers for this pull request are: @w3c/battery-status-reviewers.

dontcallmedom committed:
Remove outdated avoidance of EventHanlder / unrestricted double

dontcallmedom committed:
Wrap in an async_test (needed for Chrome?)

zqzhang commented on battery-status/battery-interface-idlharness.html:
```diff
@@ -37,15 +40,22 @@
-idl_array.add_objects({Navigator: ['navigator'], BatteryManager: ['navigator.getBattery()']});
-idl_array.test();
+t.step(function() {
+navigator.getBattery().then(function(bm) {
```
This assumes that 'navigator.getBattery()' is supported. If it is not supported, the test may return a timeout or behave like hang. Better add an assertion before this line?

Please also make proper indent.

zqzhang replied:
For example, `assert_idl_attribute(navigator, 'getBattery', 'navigator must have getBattery attribute');` in https://github.com/w3c/web-platform-tests/blob/master/battery-status/battery-interface.js#L14

zqzhang commented:
LGTM with nits.

dontcallmedom committed:
Handle error cases in using navigator.getBattery

per @zqzhang feedback https://github.com/w3c/web-platform-tests/pull/2469/files#r49553751
